packages/flutter/test/widgets/reparent_state_with_layout_builder_test.dart

```diff
@@ -69,4 +75,79 @@ void main() {
     await tester.pump();
     expect(StatefulCreationCounterState.creationCount, 1);
   });
+
+  testWidgets('Clean then reparent with dependencies',
+      (WidgetTester tester) async {
+    GlobalKey key = new GlobalKey();
+
+    StateSetter keyedSetState;
+
+    Widget keyedWidget = new StatefulBuilder(
+      key: key,
+      builder: (BuildContext context, StateSetter setState) {
+        keyedSetState = setState;
+        MediaQuery.of(context);
+        return new Container();
+      },
+    );
+
+    Widget layoutBuilderChild = keyedWidget;
+    StateSetter layoutBuilderSetState;
+
+    StateSetter childSetState;
+    Widget deepChild = new Container();
+
+    int layoutBuilderBuildCount = 0;
+
+    await tester.pumpWidget(new MediaQuery(
+      data: new MediaQueryData.fromWindow(ui.window),
+      child: new Column(
+        children: <Widget>[
+          new StatefulBuilder(
+              builder: (BuildContext context, StateSetter setState) {
+            layoutBuilderSetState = setState;
+            return new LayoutBuilder(
+              builder: (BuildContext context, BoxConstraints constraints) {
+                ++layoutBuilderBuildCount;
+                return layoutBuilderChild;
+              },
+            );
+          }),
+          new Container(
+            child: new Container(
+              child: new Container(
+                child: new Container(
+                  child: new Container(
+                    child: new Container(
+                      child: new StatefulBuilder(builder:
+                          (BuildContext context, StateSetter setState) {
+                        childSetState = setState;
+                        return deepChild;
+                      }),
+                    ),
+                  ),
+                ),
+              ),
+            ),
+          ),
+        ],
+      ),
+    ));
+
+    expect(layoutBuilderBuildCount, 1);
+
+    keyedSetState(() {});
+
+    layoutBuilderSetState(() {
+      layoutBuilderChild = new Container();
+    });
+
+    childSetState(() {
+      deepChild = keyedWidget;
+    });
+
+    await tester.pump();
+
+    expect(layoutBuilderBuildCount, 2);
+  });
 }
```

Comment:
did you try the variant of this test that doesn't have the layout builder's setState call, with your fix?